planner/core/common_plans.go

```diff
@@ -248,6 +251,12 @@ func (e *Execute) rebuildRange(p Plan) error {
 				return errors.Trace(err)
 			}
 		}
+	case *Insert:
+		return e.rebuildRange(x.SelectPlan)
+	case *Update:
+		return e.rebuildRange(x.SelectPlan)
+	case *Delete:
```

Comment:
Rebuild `SelectPlan` is not complete for `Insert` and `Update` IMHO, for example, we may have `?` in `Insert.Lists` or `Update.OrderedList`.

Reply:
It is enough to only rebuild the range for ``SelectPlan``. Since  ``?`` is expressed to a ``Constant`` with the deferred function ``GetParam()`` (see [1]), the deferred function is called whenever evaluating the expressions in ``Insert.Lists`` or ``Update.OrderedList`` (see [2]). 

[1] https://github.com/pingcap/tidb/blob/a37417e72c662367a805cc51a27628d765d68a58/planner/core/expression_rewriter.go#L763-L766

[2]
https://github.com/pingcap/tidb/blob/a37417e72c662367a805cc51a27628d765d68a58/executor/insert_common.go#L229-L230